### PR TITLE
DDF-1197 created an HTTP to HTTPS proxy to use for legacy clients

### PR DESCRIPTION
**Pull request by stustison**
- added new HTTP to HTTPS camel proxy service
 - added integration test for anonymous SOAP HTTP access
 - disabled HTTP in pax web configuration file

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf-platform/48)

### CODE REVIEW

**Comment by stustison** [comment]
@pklinef @lessarderic

File: platform-http-proxy/pom.xml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                                        </limit>
+                                        <limit>
+                                            <counter>COMPLEXITY</counter>
+                                            <value>COVEREDRATIO</value>
+                                            <minimum>0.75</minimum>
+                                        </limit>
+                                        <limit>
+                                            <counter>LINE</counter>
+                                            <value>COVEREDRATIO</value>
+                                            <minimum>0.70</minimum>
```
Comment (pklinef) [inline platform-http-proxy/pom.xml]: Why is this value 70 instead of 75?

Comment (stustison) [inline platform-http-proxy/pom.xml]: Because there were 2 lines that I couldn't test. There is an integration test that proves this all works.

**Comment by stustison** [comment]
@shaundmorris

**Comment by lessarderic** [comment]
Review status: 0 of 8 files reviewed, 5 unresolved discussions, some commit checks broke.

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 70 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiAvma9d65ifKs0t2Q)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L70)):</sup>
Odd formatting here. This is how our current formatter wraps lines?

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 71 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiCDJ3vNzhvk0XUMGM)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L71)):</sup>
I assume isHttpEnabled indicates whether the destination is https only or not?

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 74 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiC_P41f-kzbUBxBM0)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L74)):</sup>
What should we do if one or both of these are missing? Should we use default values or fail? Should the getProperties() method checks to make sure those are configured if they are required?

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 106 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiE3LGfzNO4TDLqUz8)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L106)):</sup>
If we only log here, the Properties object will be empty. Isn't this going to cause all sorts of problems down the line? Shouldn't we throw some sort of exception?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/48)

**Comment by lessarderic** [comment]
Review status: 6 of 8 files reviewed, 12 unresolved discussions, some commit checks broke.
Reviewed files:
- platform-app/src/main/resources/features.xml @ r1
- platform-http-proxy/pom.xml @ r1
- platform-http-proxy/src/main/resources/OSGI-INF/blueprint/blueprint.xml @ r1
- platform-http-proxy/src/main/resources/OSGI-INF/metatype/metatype.xml @ r1
- platform-http-proxy/src/test/resources/etc/org.ops4j.pax.web.cfg @ r1
- pom.xml @ r1

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 97 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiJnZurKhta3kEiA2X)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L97)):</sup>
Any reason this is protected, other than for unit testing?

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 39 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiG_EfI9JkjezOcs4j)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L39)):</sup>
I'm not sure what this method is testing :-( Could we have one or two simple test cases to test the basic scenarios (start/stop, invalid or missing parameters or system properties, etc.) and a separate one for this more complex test with a better method name or comment?  This would make it easier to understand what's being tested and what scenario failed when we get a unit test failure.

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 42 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiI63kfpm-mJ6HGlc9)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L42)):</sup>
Can we use actual values here (and below) instead of any*()? Would make the test a little more robust.

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 48 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiGDCWtCD7BHCvQBVO)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L48)):</sup>
Instead of re-implementing the getProperties(), can't we just set the system properties we need? That would allow us to get some coverage on the real method and would also allow us to test the scenarios where some of the system properties are missing.

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 63 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiGy5fXlLcYeNmQSPO)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L63)):</sup>
You shouldn't need to catch and call fail(). The exception should trickle up and the JUnit runner should fail the test case for you and output the information.

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 111 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiKhhQtHxlY-4PyWqx)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L111)):</sup>
Don't think we need this test case if we don't override the getProperties() method in the other test methods. That would allow us to keep that method private.

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 117 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiKSDMm26FnCSWVJFu)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L117)):</sup>
We should have a few tests for the exception scenarios.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/48)

**Comment by lessarderic** [comment]
Review status: 6 of 8 files reviewed, 12 unresolved discussions, some commit checks broke.

---

<sup>**[platform-http-proxy/pom.xml, line 110 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JphHW7MBw-9zCrNBiDw-r1-110)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/pom.xml#L110)):</sup>
Can we hit the 75% mark if we just add a couple simple unit tests for the exception scenarios?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/48)

**Comment by shaundmorris** [comment]
Review status: 6 of 8 files reviewed, 13 unresolved discussions, some commit checks broke.

---

<sup>**[platform-http-proxy/pom.xml, line 40 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiM2ymSdkrgBxRsRDx)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/pom.xml#L40)):</sup>
Should this be ${project.version} ?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/48)

**Comment by stustison** [comment]
Review status: 6 of 8 files reviewed, 11 unresolved discussions, some commit checks broke.

---

<sup>**[platform-http-proxy/pom.xml, line 40 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiM2ymSdkrgBxRsRDx)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/pom.xml#L40)):</sup>
Uh, sure :)

---

<sup>**[platform-http-proxy/pom.xml, line 110 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JphHW7MBw-9zCrNBiDw-r1-110)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/pom.xml#L110)):</sup>
I guess I could mock that when calling start or stop if it errors for some reason, I catch it? Even if I didn't catch it though, blueprint would and it'd still just log and continue. Maybe it would be better to just not even have the catch blocks and allow blueprint to snag it, then if someone configures this incorrectly it will just leave the bundle in a failed state.

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 70 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiAvma9d65ifKs0t2Q)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L70)):</sup>
Yeah I guess, I didn't format anything by hand.

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 71 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiCDJ3vNzhvk0XUMGM)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L71)):</sup>
This is to make sure that we don't try to run the proxy if someone has enabled http in pax web for some reason and tries to run this proxy. It will cause jetty to blow up.

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 74 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiC_P41f-kzbUBxBM0)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L74)):</sup>
These can't be missing, they are required by the pax web configuration. I suppose we could run into some issue with it not being able to read the pax web file maybe, but then the booleans will both equate to false and you won't fall into this block anyways. You'd also not be able to start the container.

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 97 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiJnZurKhta3kEiA2X)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L97)):</sup>
Yeah just unit testing.

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 106 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiE3LGfzNO4TDLqUz8)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L106)):</sup>
It will just fall through and not start the proxy. I can add a test for that case.

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 39 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiG_EfI9JkjezOcs4j)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L39)):</sup>
I did forget to change the name of this method when I changed how the proxy worked. The test is basically just running through the possible cases that could happen while the proxy has been running.

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 42 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiI63kfpm-mJ6HGlc9)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L42)):</sup>
The values other than the last two are completely up to however the admin has configured the system. I can drop in the last two values though.

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 48 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiGDCWtCD7BHCvQBVO)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L48)):</sup>
The issue with reading stuff from the file system is that it will not work when someone is trying to run or debug the tests from an ide. From maven, it will work fine, but then you're left with a test that you can't debug. Personally, I'd rather overwrite the method and allow a developer to debug this if there is an issue, rather than leave it in a state where it only works from maven. I'm open to suggestions though :) (I did try it that way first to see if I can make it work before I did this, just fyi)

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 63 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiGy5fXlLcYeNmQSPO)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L63)):</sup>
OK.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/48)

**Comment by lessarderic** [comment]
Review status: 5 of 8 files reviewed, 7 unresolved discussions, some commit checks pending.

---

<sup>**[platform-http-proxy/pom.xml, line 110 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JphHW7MBw-9zCrNBiDw-r1-110)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/pom.xml#L110)):</sup>
I'm fine either way, as long we unit test the solution :-)

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 93 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiJnZurKhta3kEiA2X)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L93)):</sup>
Since we're not expecting people to sub-class and override this method, can we make it package-private instead and add a note in the comment that says it's there only for unit testing? And if you decide to change the way you "mock" the properties (see my comment below), you could make this method private since you wouldn't need to override it anymore ;-)

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 102 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiE3LGfzNO4TDLqUz8)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L102)):</sup>
A test case would be nice. I also think that throwing an exception instead of leaving the Properties map empty and assuming that the calling method will handle that properly would be safer and clearer.

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 40 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiG_EfI9JkjezOcs4j)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L40)):</sup>
Thanks for updating the name, it helps some :-)

Could we split this up into two test cases:
- start/stop without a hostname
- start/stop with hostname

We should also have test cases for all the 4 SECURE_ENABLED_PROPERTY/HTTP_ENABLED_PROPERTY flag combinations and for the scenario where the getProperties() throws and exception or returns an empty Properties object.

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 42 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiI63kfpm-mJ6HGlc9)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L42)):</sup>
Those values are set in your test config file, so you should know what values to expect, right? Not matching those value is like missing asserts so I would try to use actual values here as much as possible.

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 45 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiGDCWtCD7BHCvQBVO)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L45)):</sup>
Good enough for now, but this would be an excellent place to use PowerMock and mock out the Properties and File objects. Or you could have package-private methods (say newProperties() and newFile()) that you override here and return Mockito mock objects. Then you can mock the load() and all the getProperty() calls to return what you want and you wouldn't even need to have a test config file anymore. Also makes it easier to test edge cases, e.g., missing properties or properties with invalid values.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/48)

**Comment by stustison** [comment]
Review status: 5 of 8 files reviewed, 7 unresolved discussions, some commit checks pending.

---

<sup>**[platform-http-proxy/pom.xml, line 75 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JphHW7MBw-9zCrNBiDw-r1-110)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/pom.xml#L75)):</sup>
I just removed the try/catch blocks. It will just fail the bundle if the proxy errors.

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 93 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiJnZurKhta3kEiA2X)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L93)):</sup>
I can make it package-private.

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 102 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiE3LGfzNO4TDLqUz8)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L102)):</sup>
I added the test case.

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 40 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiG_EfI9JkjezOcs4j)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L40)):</sup>
I added the test for the empty properties object.

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 42 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiI63kfpm-mJ6HGlc9)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L42)):</sup>
Not the hostname. That is entirely up to whatever machine you're running on. I added in the other 3 values.

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 45 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiGDCWtCD7BHCvQBVO)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L45)):</sup>
I'd like to avoid using PowerMock. I don't see it adding a lot of value that isn't already covered by the fact that we have an integration test that proves that the proxy works.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/48)

**Comment by lessarderic** [comment]
Looks good. Just remember to remove the jacoco config if you can. Thanks for your patience :-)

---

Review status: all files reviewed, 2 unresolved discussions, some commit checks broke.
Reviewed files:
- platform-http-proxy/pom.xml @ r2
- platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java @ r2
- platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java @ r2

---

<sup>**[platform-http-proxy/pom.xml, line 110 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JphHW7MBw-9zCrNBiDw-r1-110)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/pom.xml#L110)):</sup>
Are we at 75% now? Can we remove this jacoco config block? :-)

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/48)

**Comment by stustison** [comment]
I did remove it. If you check, it should be gone.

---

Review status: 7 of 8 files reviewed, 2 unresolved discussions, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/48)

**Comment by lessarderic** [comment]
Review status: all files reviewed, 2 unresolved discussions, some commit checks pending.
Reviewed files:
- platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java @ r3

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/48)

**Comment by stustison** [comment]
Review status: all files reviewed, 2 unresolved discussions, some commit checks pending.

---

<sup>**[platform-http-proxy/pom.xml, line 40 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JpiM2ymSdkrgBxRsRDx)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/pom.xml#L40)):</sup>
Done.

---

<sup>**[platform-http-proxy/pom.xml, line 75 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-JphHW7MBw-9zCrNBiDw-r1-110)** ([raw file](https://github.com/codice/ddf-platform/blob/7fcad29afd0353c86fe2c137810e5098fcab4068/platform-http-proxy/pom.xml#L75)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/48)

**Comment by pklinef** [comment]
We should add documentation to either platform or distro to explain how and when to use this proxy.

**Comment by pklinef** [comment]
Review status: all files reviewed, 3 unresolved discussions, some commit checks broke.

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 45 \[r3\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-Jpo0ETGcTJBBwN9yLtr)** ([raw file](https://github.com/codice/ddf-platform/blob/65fb3619de06e2d7879c3253a3f45355ada9e050/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L45)):</sup>
If you change the system property for karaf home to be the test resource directory, you can avoid having to do this and increase your test coverage.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/48)

**Comment by stustison** [comment]
Review status: all files reviewed, 3 unresolved discussions, some commit checks broke.

---

<sup>**[platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java, line 45 \[r3\]](https://reviewable.io:443/reviews/codice/ddf-platform/48#-Jpo0ETGcTJBBwN9yLtr)** ([raw file](https://github.com/codice/ddf-platform/blob/65fb3619de06e2d7879c3253a3f45355ada9e050/platform-http-proxy/src/test/java/org/codice/ddf/platform/http/proxy/TestHttpProxy.java#L45)):</sup>
That only works if you're running from maven, not from an ide.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/48)

**Comment by pklinef** [comment]
:+1:

**Comment by stustison** [comment]
I agree, I'll make a ticket to flesh out some documentation on how and when to use the proxy.

---

Review status: all files reviewed, 3 unresolved discussions, some commit checks broke.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/48)
